tenant/storage.go

```diff
@@ -13,15 +14,30 @@ const MaxIDGenerationN = 100
 const ReservedIDs = 1000
 
 type Store struct {
-	kvStore kv.Store
-	IDGen   influxdb.IDGenerator
+	kvStore        kv.Store
+	IDGen          influxdb.IDGenerator
+	urmByUserIndex *kv.Index
 }
 
 func NewStore(kvStore kv.Store) (*Store, error) {
 	st := &Store{
 		kvStore: kvStore,
 		IDGen:   snowflake.NewDefaultIDGenerator(),
+		urmByUserIndex: kv.NewIndex(kv.NewIndexMapping(
+			urmBucket,
+			urmByUserIndexBucket,
+			func(v []byte) ([]byte, error) {
+				var urm influxdb.UserResourceMapping
+				if err := json.Unmarshal(v, &urm); err != nil {
+					return nil, err
+				}
+
+				id, _ := urm.UserID.Encode()
+				return id, nil
+			},
+		)),
 	}
+	kv.WithIndexReadPathEnabled(st.urmByUserIndex)
```

Comment:
You can actually pass this as a variadic option, I was just being weird in my previous implementation because of how I wired in the configuration.
e.g. you can:
```go
urmByUserIndex: kv.NewIndex(kv.NewIndexMapping(
    urmBucket,
    urmByUserIndexBucket,
    func(v []byte) ([]byte, error) {
        var urm influxdb.UserResourceMapping
        if err := json.Unmarshal(v, &urm); err != nil {
            return nil, err
        }

        id, _ := urm.UserID.Encode()
        return id, nil
    },
), kv.WithIndexReadPathEnabled),
```

Reply:
oo nice. thanks!